virtualbox.py

```diff
@@ -0,0 +1,78 @@
+import subprocess
+from pathlib import Path
+from sys import platform
+import datetime
+from pytz import UTC
+
+utc = UTC
+
+class virtualbox(object):
+
+    def __init__(self, virtualboxpath, tz) -> None:
+        if platform.lower() == "win32":
+            virtualboxpath = virtualboxpath.replace('\"', "")
+            virtualboxpath = virtualboxpath.replace("\'", "")
+            self.vmrunpath = Path(virtualboxpath).joinpath("VBoxManage.exe")
+        else:
+            self.vmrunpath = virtualboxpath
+        self.tz = tz
+        self.vminfo = []
+        self.output = []
+
+    def updateOutput(self) -> None:
+        # output = subprocess.run([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE)
+        # output = output.stdout.decode("utf-8")
+        with subprocess.Popen([str(self.vmrunpath), "list", "runningvms"], stdout=subprocess.PIPE) as proc:
+            output = proc.stdout.read().decode()
+        if platform.lower() == "win32":
+            output = output.split("\r\n")
+        else:
+            output = output.split("\n")
+        self.output = [{"Name": x[:x.find(' {')].strip().replace('"', ''), "Hash": x[x.find(' {'):].strip()} for x in output if len(x)]
+        self.vminfo = [[] for _ in range(self.runCount())]
+        for i in range(self.runCount()):
+            # vminfo = subprocess.run([str(self.vmrunpath), "showvminfo", self.getGuestName(i)], stdout=subprocess.PIPE)
+            # vminfo = vminfo.stdout.decode("utf-8")
+            with subprocess.Popen([str(self.vmrunpath), "showvminfo", self.getGuestName(i)], stdout=subprocess.PIPE) as proc:
+                vminfo = proc.stdout.read().decode()
+            vminfo = vminfo.replace("Name", "displayName", 1)
```

Comment:
Why?

Reply:
In case a user assigned a shared folder between a VM and the host this produces another attribute called "Name" in the output which is hard to work with without loosing the ability to access most of the attributes (by stripping the end of the output) or a more complex filter